### DISCUSSION TITLE
[react] Use React's `useId` hook once React 18 is live

### DISCUSSION THREAD

**Discussion by MichaelDeBoey** [tailwindlabs/headlessui#1009] [Ideas]
Since the `useId` hook will be available in React 18, the [custom `useId` hook](https://github.com/tailwindlabs/headlessui/blob/e9e6aded545c329585d8cf7452ad9fe400a5406b/packages/%40headlessui-react/src/hooks/use-id.ts) can be deprecated once it's live.

Ref: https://github.com/reactwg/react-18/discussions/111
Ref: https://reactjs.org/docs/hooks-reference.html#useid

**Comment by lailo** [comment]
I guess it's time to start using the react useId hook 🚀

**Comment by RobinMalfait** [comment]
We now use this hook in the latest `insiders` build if you are on React 18!
